internal/pkg/warewulfd/util_test.go

```diff
@@ -2,34 +2,40 @@ package warewulfd
 
 import (
 	"testing"
-	"github.com/stretchr/testify/assert"
+
 	warewulfconf "github.com/hpcng/warewulf/internal/pkg/config"
+	"github.com/hpcng/warewulf/internal/pkg/node"
+	"github.com/hpcng/warewulf/internal/pkg/wwlog"
+	"github.com/stretchr/testify/assert"
 )
 
-var getOverlayFileTests = []struct{
+var getOverlayFileTests = []struct {
 	description string
-	node string
-	context string
-	overlays []string
-	result string
-	succeed bool
+	node        string
+	context     string
+	overlays    []string
+	result      string
+	succeed     bool // getOverlayFile will always fail if no overlay dir is defined!
 }{
-	{"empty", "", "", nil, "", true},
-	{"empty node", "node1", "", nil, "", true},
-	{"specific overlays without node", "", "", []string{"o1", "o2"}, "p/overlays/o1-o2.img", true},
-	{"system overlay", "node1", "system", nil, "p/overlays/node1/__SYSTEM__.img", true},
-	{"runtime overlay", "node1", "runtime", nil, "p/overlays/node1/__RUNTIME__.img", true},
-	{"specific overlay", "node1", "", []string{"o1"}, "p/overlays/node1/o1.img", true},
-	{"multiple specific overlays", "node1", "", []string{"o1", "o2"}, "p/overlays/node1/o1-o2.img", true},
-	{"multiple specific overlays with context", "node1", "system", []string{"o1", "o2"}, "p/overlays/node1/__SYSTEM__.img", true},
+	{"empty", "", "", nil, "", false},
+	{"empty node", "node1", "", nil, "", false},
+	{"specific overlays without node", "", "", []string{"o1", "o2"}, "p/overlays/node1/o1-o2.img", false},
+	{"system overlay", "node1", "system", nil, "p/overlays/node1/__SYSTEM__.img", false},
+	{"runtime overlay", "node1", "runtime", nil, "p/overlays/node1/__RUNTIME__.img", false},
+	{"specific overlay", "node1", "", []string{"o1"}, "p/overlays/node1/o1.img", false},
+	{"multiple specific overlays", "node1", "", []string{"o1", "o2"}, "p/overlays/node1/o1-o2.img", false},
+	{"multiple specific overlays with context", "node1", "system", []string{"o1", "o2"}, "p/overlays/node1/__SYSTEM__.img", false},
 }
 
 func Test_getOverlayFile(t *testing.T) {
+	wwlog.SetLogLevel(wwlog.VERBOSE)
 	conf := warewulfconf.Get()
 	conf.Paths.WWProvisiondir = "p"
+	var n node.NodeInfo
```

Comment:
This should be defined within the `for` loop to ensure we don't leak data from one test to another.